src/index.ts

```diff
@@ -15,27 +29,152 @@ function traverseFilePathUntil(filename: string, predicate: (filename: string) =
+function findRoot(): string {
+  return (
+    process.env.OCLIF_TEST_ROOT ??
+    // eslint-disable-next-line unicorn/prefer-module
+    Object.values(require.cache).find((m) => m?.children.includes(module))?.filename ??
+    traverseFilePathUntil(
+      // eslint-disable-next-line unicorn/prefer-module
+      require.main?.path ?? module.path,
+      (p) => !(p.includes('node_modules') || p.includes('.pnpm') || p.includes('.yarn')),
+    )
   )
-/* eslint-enable unicorn/prefer-module */
+}
+
+function makeLoadOptions(loadOpts?: Interfaces.LoadOptions): Interfaces.LoadOptions {
+  return loadOpts ?? {root: findRoot()}
+}
+
+/**
+ * Capture the stderr and stdout output of a function
+ * @param fn async function to run
+ * @param opts options
+ *  - print: Whether to print the output to the console
+ *  - stripAnsi: Whether to strip ANSI codes from the output
+ * @returns {Promise<CaptureResult<T>>} Captured output
+ *   - error: Error object if the function throws an error
+ *   - result: Result of the function if it returns a value and succeeds
+ *   - stderr: Captured stderr output
+ *   - stdout: Captured stdout output
+ */
+export async function captureOutput<T>(fn: () => Promise<unknown>, opts?: CaptureOptions): Promise<CaptureResult<T>> {
+  const print = opts?.print ?? false
+  const stripAnsi = opts?.stripAnsi ?? true
+
+  const originals = {
+    NODE_ENV: process.env.NODE_ENV,
+    stderr: process.stderr.write,
+    stdout: process.stdout.write,
+  }
+
+  const output: Record<'stderr' | 'stdout', Array<Uint8Array | string>> = {
+    stderr: [],
+    stdout: [],
+  }
+
+  const toString = (str: Uint8Array | string): string => (stripAnsi ? ansis.strip(str.toString()) : str.toString())
+  const getStderr = (): string => output.stderr.map((b) => toString(b)).join('')
+  const getStdout = (): string => output.stdout.map((b) => toString(b)).join('')
 
-// Using a named export to import fancy causes this issue: https://github.com/oclif/test/issues/516
-export const test = fancyTest.fancy
-  .register('loadConfig', loadConfig)
-  .register('command', command)
-  .register('exit', exit)
-  .register('hook', hook)
-  .env({NODE_ENV: 'test'})
+  const mock =
+    (std: 'stderr' | 'stdout'): MockedStderr | MockedStdout =>
+    (str: Uint8Array | string, encoding?: ((err?: Error) => void) | BufferEncoding, cb?: (err?: Error) => void) => {
+      output[std].push(str)
 
-export default test
+      if (print) {
+        if (encoding !== null && typeof encoding === 'function') {
+          cb = encoding
+          encoding = undefined
+        }
 
-export {command} from './command'
+        originals[std].apply(process[std], [str, encoding, cb])
+      } else if (typeof cb === 'function') cb()
 
-export {Config} from '@oclif/core'
-export {FancyTypes, expect} from 'fancy-test'
+      return true
+    }
+
+  process.stdout.write = mock('stdout')
+  process.stderr.write = mock('stderr')
+  process.env.NODE_ENV = 'test'
```

Comment:
The current major does this too: https://github.com/oclif/test/blob/main/src/index.ts#L34

`NODE_ENV` needs to be `test` or `development` in order for things like devPlugins to be loaded. So, for instance, plugin-not-found has plugin-commands as a devPlugin so that we can test that misspelling the `commands` command is properly handled.